pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -740,20 +742,44 @@ func (r *ReconcileClusterSync) applyPatch(
 	return nil, false
 }
 
+// ensureHiveSourceAnnotation makes sure obj has a HiveSourcesAnnotation containing an
+// appropriate value for syncSet.
+// Currently the annotation is a space-separated list of strings of the form Kind/Name,
+// where Kind is either SyncSet or SelectorSyncSet, and Name is the name of the
+// [Selector]SyncSet.
+func ensureHiveSourceAnnotation(obj hivev1.MetaRuntimeObject, syncSet CommonSyncSet) {
```

Comment:
This is not going to work. The `obj` passed into this function will not contain the annotations that exist in the target cluster. If there are multiple syncsets applying the same resource to the target cluster, they will each use a different value for the annotation: The annotation values will not get combined.

Reply:
Well dang. I'll have to dig for a way around that.

In general, are you supportive of this as a concept?

Reply:
I do not find the concept objectionable, but I am not convinced of the utility or whether we want to expose these details in customer-accessible objects. My biggest concern is that I think it is going to be very difficult to get it right.

Reply:
That's fair. It's not *hugely* more convenient than searching through the ClusterSync. If it was easy -- if it would work as written -- then it might be worth it; but if it's going to be hard...